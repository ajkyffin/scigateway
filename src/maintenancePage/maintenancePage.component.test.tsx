import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import {
  CombinedMaintenancePageProps,
  MaintenancePageWithStyles,
} from './maintenancePage.component';

const dummyClasses = {
  root: 'root-class',
  container: 'container-class',
  titleText: 'titleText-class',
};

describe('Maintenance page component', () => {
  let mount;
  let props: CombinedMaintenancePageProps;

  beforeEach(() => {
    mount = createMount();

    props = {
      message: 'test',
      classes: dummyClasses,
    };
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('should render correctly', () => {
    const wrapper = mount(<MaintenancePageWithStyles {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
