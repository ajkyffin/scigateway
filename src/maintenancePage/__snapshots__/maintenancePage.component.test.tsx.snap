// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maintenance page component should render correctly 1`] = `
<WithStyles(MaintenancePage)
  classes={
    Object {
      "container": "container-class",
      "root": "root-class",
      "titleText": "titleText-class",
    }
  }
  message="test"
>
  <MaintenancePage
    classes={
      Object {
        "container": "MaintenancePage-container-2 container-class",
        "root": "MaintenancePage-root-1 root-class",
        "titleText": "MaintenancePage-titleText-3 titleText-class",
      }
    }
    message="test"
  >
    <div
      className="MaintenancePage-root-1 root-class"
    >
      <WithStyles(ForwardRef(Typography))
        className="MaintenancePage-titleText-3 titleText-class"
        variant="h3"
      >
        <ForwardRef(Typography)
          className="MaintenancePage-titleText-3 titleText-class"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          variant="h3"
        >
          <h3
            className="MuiTypography-root MaintenancePage-titleText-3 titleText-class MuiTypography-h3"
          >
            Maintenance
          </h3>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
      <div
        className="MaintenancePage-container-2 container-class"
      >
        <WithStyles(ForwardRef(Typography))
          variant="h4"
        >
          <ForwardRef(Typography)
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
            variant="h4"
          >
            <h4
              className="MuiTypography-root MuiTypography-h4"
            >
              test
            </h4>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
  </MaintenancePage>
</WithStyles(MaintenancePage)>
`;
